src/control/common/env_utils_test.go

```diff
@@ -0,0 +1,152 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package common
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestScrubEnvironment(t *testing.T) {
+	defaultSetup := []string{
+		"FOO=bar", "baz=quux", "COW=QUACK", "ANSWER=42",
+	}
+	for name, tc := range map[string]struct {
+		setup     []string
+		list      []string
+		whitelist bool
+		expected  []string
+	}{
+		"nil blacklist": {},
+		"empty blacklist": {
+			list: []string{},
+		},
+		"blacklist": {
+			list:     []string{"COW", "baz"},
+			expected: []string{"FOO=bar", "ANSWER=42"},
+		},
+		"nil whitelist": {
+			whitelist: true,
+			expected:  []string{},
+		},
+		"empty whitelist": {
+			list:      []string{},
+			whitelist: true,
+			expected:  []string{},
+		},
+		"whitelist": {
+			list:      []string{"ANSWER"},
+			whitelist: true,
+			expected:  []string{"ANSWER=42"},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			os.Clearenv()
```

Comment:
Should we be restoring the original environment after the test is done?

Reply:
The tests only modify their own environment, not the system environment. I'd also argue that any test that's depending on the environment to be set up a certain way should make that happen. That having been said, it's simple to capture/restore the environment so I'll do that.